src/steam_network/user_info_cache.py

```diff
@@ -31,29 +31,31 @@ def is_initialized(self) -> bool:
 
 
     def to_dict(self):
-        creds = {
-            'steam_id': base64.b64encode(str(self._steam_id).encode('utf-8')).decode('utf-8'),
-            'refresh_token': base64.b64encode(str(self._refresh_token).encode('utf-8')).decode('utf-8'),
-            'account_username': base64.b64encode(str(self._account_username).encode('utf-8')).decode('utf-8'),
-            'persona_name': base64.b64encode(str(self._persona_name).encode('utf-8')).decode('utf-8'),
-        }
+        creds = {}
+        if self.is_initialized():
+            creds = {
+                'steam_id': base64.b64encode(str(self._steam_id).encode('utf-8')).decode('utf-8'),
+                'refresh_token': base64.b64encode(str(self._refresh_token).encode('utf-8')).decode('utf-8'),
+                'account_username': base64.b64encode(str(self._account_username).encode('utf-8')).decode('utf-8'),
+                'persona_name': base64.b64encode(str(self._persona_name).encode('utf-8')).decode('utf-8'),
+            }
         return creds
 
     def from_dict(self, lookup):
         for key in lookup.keys():
             if lookup[key]:
                 logger.info(f"Loaded {key} from stored credentials")
 
-        if 'steam_id' in lookup:
+        if 'steam_id' in lookup and lookup['steam_id'] is not None:
```

Comment:
*the following applies to lines 49, 52, 55, 58:*

The same could be achieved with just a `if lookup.get('steam_id') is not None:`

Reply:
its not another index op though? dict.get(key) basically is a dict[key] except if the dict doesn't have the key it returns a default value (which is by default `None`) and you're checking if the value itself is not None anyways. seems pretty 1:1 to me.